src/odc/core/MetaData.cc

```diff
@@ -87,26 +88,28 @@ Column* MetaData::columnByName(const std::string& name) const
 bool MetaData::hasColumn(const std::string& name) const
 { 
 	for (size_t i = 0; i < size(); i++)
-		if (at(i)->name() == name || at(i)->name().find(name + "@") == 0)
+		if (columnNameMatches(at(i)->name(), name))
 			return true;
 	return false;
 }
 
 size_t MetaData::columnIndex(const std::string& name) const
 {
-	std::vector<size_t> indices;
+	const size_t notFound = static_cast<size_t>(-1);
+	size_t index = notFound;
 
 	for (size_t i = 0; i < size(); i++)
-		if (at(i)->name() == name || at(i)->name().find(name + "@") == 0)
-			indices.push_back(i);
-
-	if (indices.size() > 1)
-		throw eckit::UserError(std::string("Ambiguous column name: '") + name + "'");
+		if (columnNameMatches(at(i)->name(), name)) {
+			if (index == notFound)
+				index = i;
+			else
+				throw eckit::UserError(std::string("Ambiguous column name: '") + name + "'");
```

Comment:
Please create a derived exception type (see comment below). Something along the lines of:
class AmbiguousColumnException : public UserError {
    AmbiguousColumnException(const std::string& columnName) { ...}
    ....
};
This lets us do the exception wrapping that you do below more cleanly.